app/base_api_error.py

```diff
@@ -0,0 +1,12 @@
+from flask import jsonify
+
+
+class BaseApiError(Exception):
+    def __init__(self, message, error_code):
+        super().__init__(message)
+        self.error_code = error_code
+
+    def buildError(self):
+        return jsonify(
+            {"success": False, "message": self.message, "error_code": self.error_code}

```

Comment:
If we want to use this as something we can return from our endpoint functions, the error code should probably be returned outside of the actual body. For example, change to this: `return jsonify({"success": False, "message": self.message}), self.error_code`